typed/typed.go

```diff
@@ -187,7 +201,37 @@ func (tv TypedValue) RemoveItems(items *fieldpath.Set) *TypedValue {
 }
 
 // ExtractItems returns a value with only the provided list or map items extracted from the value.
-func (tv TypedValue) ExtractItems(items *fieldpath.Set) *TypedValue {
+func (tv TypedValue) ExtractItems(items *fieldpath.Set, opts ...extractItemsOption) *TypedValue {
```

Comment:
```suggestion
func (tv TypedValue) ExtractItems(items *fieldpath.Set, opts ...ExtractItemsOption) *TypedValue {
```

Requiring a non-exported type as an argument of an exported type is problematic. 

One way to detect this would be to use the test package in `typed/remove_test.go`.  I don't think you need necessarily to do that here, but that's a good way to test against only the exported API.

Reply:
My original thought was that putting it non-exported can force user to only use the exported `Withxxx()` function to construct the options.  But it indeed looks strange to put it in the exported function. 

Updated.